addons/hr_holidays/models/hr_employee.py

```diff
@@ -513,7 +513,8 @@ def _get_consumed_leaves(self, leave_types, target_date=False, ignore_future=Fal
                                 datetime.combine(allocation.date_to, time.max)
                                 if allocation.date_to else leave.date_to
                             ).replace(tzinfo=pytz.UTC)
-                            duration_info = employee._get_calendar_attendances(interval_start, interval_end)
+                            employee_timezone = pytz.timezone(employee.tz) if employee.tz else None
+                            duration_info = employee.with_context(employee_timezone=employee_timezone)._get_calendar_attendances(interval_start, interval_end)
```

Comment:
Since the context is added in `_get_calendar_attendances`, this is no longer useful :)